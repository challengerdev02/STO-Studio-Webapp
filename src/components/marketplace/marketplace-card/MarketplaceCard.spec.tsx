import {
  cleanup,
  queryAllByAltText,
  queryAllByAttribute,
  queryByRole,
  queryByText,
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MarketplaceCard } from './index';

describe('MarketplaceCard', function () {
  afterEach(cleanup);
  let props = {
    coverInfo: {
      countDown: '01:52:09 left ðŸ”¥',
      startDate: new Date(2022, 0, 1, 12, 30, 20, 0).toDateString(),
      endDate: new Date(2021, 11, 16, 12, 30, 7, 999).toDateString(),
      coverImg:
        'https://robohash.org/a8cd9c187bb33c8a8f6e2774d71308fb?set=set4&bgset=&size=400x400',
      menuItems: ['Option A', 'Option B', 'Option c'],
    },
    cardData: {
      title: 'Burney #282',
      iconURL:
        'https://robohash.org/0f0383b8d198fb15c0ab7fc729a73028?set=set4&bgset=&size=400x400',
      itemPrice: '3.01 BTC',
      rates: '1/1',
      count: '17k',
      inWishlist: false,
      urlList: [
        'https://robohash.org/962440efdb9e25e8ff4360a7037acfb4?set=set4&bgset=&size=400x400',
        'https://robohash.org/887f430666b7d04d90b82a4b988df961?set=set4&bgset=&size=400x400',
        'https://robohash.org/15b76d2ec3afc53e3af6c4a316090c94?set=set4&bgset=&size=400x400',
      ],
    },
    style: {
      width: '256px',
    },
    hoverable: true,
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<MarketplaceCard {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should display the card', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryByRole(document.body, 'MarketplaceCardContainer')
    ).toBeInTheDocument();
  });

  it('should display the card coverImage', () => {
    // @ts-ignore
    const { container } = setup({ cover: 'This is test cover' });
    expect(
      queryAllByAttribute('class', document.body, 'ant-card-cover')[0]
    ).toBeInTheDocument();
    expect(
      queryAllByAttribute('class', document.body, 'ant-card-cover')[0]
    ).toHaveStyle(
      'background-image: url(https://robohash.org/a8cd9c187bb33c8a8f6e2774d71308fb?set=set4&bgset=&size=400x400);'
    );
  });

  it('should have countDown "This is a test for countDown"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(queryByRole(document.body, 'count-down-text')).toBeInTheDocument();
  });

  it('should have dropdown menu "This is a test dropdown menu"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryAllByAttribute(
        'class',
        document.body,
        'ant-btn ant-btn-default ant-btn-icon-only ant-dropdown-trigger'
      )[0]
    ).toBeInTheDocument();
    // expect(queryByText(document.body, "Option B")).toHaveClass(
    //     "ant-dropdown-menu-title-content"
    //   );
  });

  it('should have Title "This is a test Title"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(queryByText(document.body, 'Burney #282')).toBeInTheDocument();
    expect(queryByText(document.body, 'Burney #282')).toHaveClass(
      'ant-typography'
    );
  });

  it('should have StackedImages "This is a test StackImages"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryAllByAltText(document.body, 'stack-image')[0]
    ).toBeInTheDocument();
    expect(queryAllByAltText(document.body, 'stack-image')[0]).toHaveStyle(
      'width: 100%;'
    );
  });

  it('This is a test for Not in wishlist', () => {
    // @ts-ignore
    const { container } = setup({
      cardData: {
        inWishlist: false,
      },
    });
    expect(queryByRole(document.body, 'NotInWishlist')).toBeInTheDocument();
    expect(queryByRole(document.body, 'NotInWishlist')).toHaveClass(
      'anticon-heart'
    );
  });

  it('This is a test for in wishlist', () => {
    // @ts-ignore
    const { container } = setup({
      cardData: {
        inWishlist: true,
      },
    });
    expect(queryByRole(document.body, 'InWishlist')).toBeInTheDocument();
    expect(queryByRole(document.body, 'InWishlist')).toHaveClass(
      'anticon-heart'
    );
  });
});
