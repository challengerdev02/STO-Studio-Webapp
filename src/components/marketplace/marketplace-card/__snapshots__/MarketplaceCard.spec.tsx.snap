// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketplaceCard should match with snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable sc-eCImPb eXMxhc main-mc-container"
    countDown="01:52:09 left ðŸ”¥"
    coverImg="https://robohash.org/a8cd9c187bb33c8a8f6e2774d71308fb?set=set4&bgset=&size=400x400"
    enddate="Thu Dec 16 2021"
    menuItems="Option A,Option B,Option c"
    role="MarketplaceCardContainer"
    startdate="Sat Jan 01 2022"
    style="width: 256px;"
  >
    <div
      class="ant-card-cover"
    >
      <div>
        <div
          class="sc-bdvvtL dIxYVf"
        >
          <div
            class="sc-gsDKAQ cOcCgz count-down-text"
            role="count-down-text"
          >
             days,  hours left ðŸ”¥
          </div>
          <div
            class="ant-btn-group ant-dropdown-button"
          >
            <button
              class="ant-btn ant-btn-default"
              type="button"
            />
            <button
              class="ant-btn ant-btn-default ant-btn-icon-only ant-dropdown-trigger"
              type="button"
            >
              <span
                aria-label="ellipsis"
                class="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="sc-bdvvtL dIxYVf"
      >
        <span
          class="ant-typography sc-gKclnd elkVOZ"
          font-size="14px"
          font-weight="500"
        >
          Burney #282
        </span>
        <div
          class="sc-dkPtRN sc-hKwDye kToXgV rBElr"
          height="24px"
          width="24px"
        >
          <img
            alt="Burney #282"
            class="sc-iCfMLu bpDNvp"
            src="https://robohash.org/0f0383b8d198fb15c0ab7fc729a73028?set=set4&bgset=&size=400x400"
            width="100%"
          />
        </div>
      </div>
      <div
        class="sc-bdvvtL dIxYVf"
      >
        <span>
          <span
            class="ant-typography sc-gKclnd fILiiP"
            font-size="12px"
            font-weight="600"
            line_height="20px"
          >
            3.01 BTC
          </span>
          <span
            class="ant-typography sc-gKclnd KfwZx"
            font-size="12px"
            font-weight="500"
            margin="0 0 0 4.5px"
          >
            1/1
          </span>
        </span>
        <span>
          <span
            aria-label="heart"
            class="anticon anticon-heart"
            role="NotInWishlist"
          >
            <svg
              aria-hidden="true"
              data-icon="heart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
              />
            </svg>
          </span>
          <span
            class="ant-typography sc-gKclnd foTPFP"
            font-size="14px"
            font-weight="600"
            line_height="160%"
            margin=" 0 0 0 4.5px"
          >
            17k
          </span>
        </span>
      </div>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
      />
      <div
        class="sc-bdvvtL dIxYVf"
      >
        <div
          class="sc-bdvvtL sc-jRQBWg cxIwvg gqzvZh"
          width="auto"
        >
          <div
            class="sc-dkPtRN sc-hKwDye kToXgV rBElr stacked-img"
            height="24px"
            role="stackImages"
            width="24px"
          >
            <img
              alt="stack-image"
              class="sc-iCfMLu bpDNvp"
              src="https://robohash.org/962440efdb9e25e8ff4360a7037acfb4?set=set4&bgset=&size=400x400"
              width="100%"
            />
          </div>
          <div
            class="sc-dkPtRN sc-hKwDye kToXgV kFSgPO stacked-img"
            height="24px"
            role="stackImages"
            width="24px"
          >
            <img
              alt="stack-image"
              class="sc-iCfMLu bpDNvp"
              src="https://robohash.org/887f430666b7d04d90b82a4b988df961?set=set4&bgset=&size=400x400"
              width="100%"
            />
          </div>
          <div
            class="sc-dkPtRN sc-hKwDye kToXgV heHuFZ stacked-img"
            height="24px"
            role="stackImages"
            width="24px"
          >
            <img
              alt="stack-image"
              class="sc-iCfMLu bpDNvp"
              src="https://robohash.org/15b76d2ec3afc53e3af6c4a316090c94?set=set4&bgset=&size=400x400"
              width="100%"
            />
          </div>
        </div>
        <span>
          <button
            class="ant-btn ant-btn-link"
            style="padding: 0px; font-size: 14px; font-weight: 600; margin: 0px 0px 0px 4.5px; line-height: 160%;"
            type="button"
          >
            <span
              aria-label="sliders"
              class="anticon anticon-sliders"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="sliders"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M320 224h-66v-56c0-4.4-3.6-8-8-8h-52c-4.4 0-8 3.6-8 8v56h-66c-4.4 0-8 3.6-8 8v560c0 4.4 3.6 8 8 8h66v56c0 4.4 3.6 8 8 8h52c4.4 0 8-3.6 8-8v-56h66c4.4 0 8-3.6 8-8V232c0-4.4-3.6-8-8-8zm-60 508h-80V292h80v440zm644-436h-66v-96c0-4.4-3.6-8-8-8h-52c-4.4 0-8 3.6-8 8v96h-66c-4.4 0-8 3.6-8 8v416c0 4.4 3.6 8 8 8h66v96c0 4.4 3.6 8 8 8h52c4.4 0 8-3.6 8-8v-96h66c4.4 0 8-3.6 8-8V304c0-4.4-3.6-8-8-8zm-60 364h-80V364h80v296zM612 404h-66V232c0-4.4-3.6-8-8-8h-52c-4.4 0-8 3.6-8 8v172h-66c-4.4 0-8 3.6-8 8v200c0 4.4 3.6 8 8 8h66v172c0 4.4 3.6 8 8 8h52c4.4 0 8-3.6 8-8V620h66c4.4 0 8-3.6 8-8V412c0-4.4-3.6-8-8-8zm-60 145a3 3 0 01-3 3h-74a3 3 0 01-3-3v-74a3 3 0 013-3h74a3 3 0 013 3v74z"
                />
              </svg>
            </span>
            <span>
              Place a bid
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;
